src/main/java/de/hysky/skyblocker/skyblock/dwarven/CorpseFinder.java

```diff
@@ -0,0 +1,269 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+
+public class CorpseFinder {
+    private static final Location LOCATION = Location.GLACITE_MINESHAFT;
+    private static boolean isLocationCorrect = false;
+    private static final Pattern CORPSE_FOUND_PATTERN = Pattern.compile("([A-Z]+) CORPSE LOOT!");
+    private static final String PREFIX = "[Skyblocker Corpse Finder] ";
+    private static final Logger LOGGER = LoggerFactory.getLogger(CorpseFinder.class);
+    private static final Map<String, List<Corpse>> corpsesByType = new HashMap<>();
+	private static final Set<String> ITEM_IDS = Set.of(
+			"LAPIS_ARMOR_HELMET",
+			"ARMOR_OF_YOG_HELMET",
+			"MINERAL_HELMET",
+			"VANGUARD_HELMET"
+	);
+
+    @Init
+    public static void init() {
+        ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> isLocationCorrect = false);
+        SkyblockEvents.LOCATION_CHANGE.register(CorpseFinder::handleLocationChange);
+        ClientReceiveMessageEvents.GAME.register(CorpseFinder::onChatMessage);
+        WorldRenderEvents.AFTER_TRANSLUCENT.register(CorpseFinder::renderWaypoints);
+        ClientTickEvents.END_CLIENT_TICK.register(client -> {
+            if (!SkyblockerConfigManager.get().mining.glacite.enableCorpseFinder || client.player == null) return;
+            if (!isLocationCorrect) return;
+            for (List<Corpse> corpses : corpsesByType.values()) {
+                for (Corpse corpse : corpses) {
+                    if (!corpse.seen && client.player.canSee(corpse.entity)) {
+                        setSeen(corpse);
+                    }}}});
+        ClientCommandRegistrationCallback.EVENT.register((dispatcher, registryAccess) -> dispatcher.register(literal(SkyblockerMod.NAMESPACE)
+            .then(literal("corpseHelper")
+                .then(literal("shareLocation")
+                    .then(argument("blockPos", ClientBlockPosArgumentType.blockPos())
+                        .then(argument("corpseType", CorpseTypeArgumentType.corpseType())
+                            .executes(context -> {
+                                MessageScheduler.INSTANCE.sendMessageAfterCooldown("/pc " + "[Skyblocker] " + "Corpse " + context.getArgument("corpseType", String.class) + " found at " + context.getArgument("blockPos", ClientPosArgument.class).toAbsoluteBlockPos(context.getSource()).toShortString());
+                                return Command.SINGLE_SUCCESS;
+                            })))))));
+    }
+
+    private static boolean seenDebugWarning = false;
+    private static void handleLocationChange(Location location) {
+        isLocationCorrect = location == LOCATION;   // true if mineshafts else false
```

Comment:
```corpseByType``` is never cleared so if you do not loot all the corpses in a mineshaft next time you go into one a mineshaft the mod will still be looking for a corpse at that location. I suggest cleaing it when entering / leaving the mineshaft (here)
```java 
corpsesByType.clear();
```

Reply:
Done in [3fb10dc](https://github.com/SkyblockerMod/Skyblocker/pull/960/commits/3fb10dc27d31732c3b0168bde2e056e0e42d70f4).